lib/lz4.h

```diff
@@ -665,18 +665,17 @@ union LZ4_streamDecode_u {
 #ifdef LZ4_DISABLE_DEPRECATE_WARNINGS
 #  define LZ4_DEPRECATED(message)   /* disable deprecation warnings */
 #else
-#  define LZ4_GCC_VERSION (__GNUC__ * 100 + __GNUC_MINOR__)
 #  if defined (__cplusplus) && (__cplusplus >= 201402) /* C++14 or greater */
 #    define LZ4_DEPRECATED(message) [[deprecated(message)]]
-#  elif (LZ4_GCC_VERSION >= 405) || defined(__clang__)
-#    define LZ4_DEPRECATED(message) __attribute__((deprecated(message)))
-#  elif (LZ4_GCC_VERSION >= 301)
-#    define LZ4_DEPRECATED(message) __attribute__((deprecated))
 #  elif defined(_MSC_VER)
 #    define LZ4_DEPRECATED(message) __declspec(deprecated(message))
+#  elif defined(__clang__) || (defined(__GNUC__) && (__GNUC__ >= 4) && (__GNUC_MINOR__ >= 5))
```

Comment:
I think `__GNUC_MINOR__` is being used incorrectly.
Wouldn't gcc 7.1.0 translate to `__GNUC__ == 7` & `__GNUC_MINOR__ == 1`, which would fail this test?

I believe it should read:
`#elif defined(__clang__) || (defined(__GNUC__) && ((__GNUC__ > 4) || ((__GNUC__ == 4) && (__GNUC_MINOR__ >= 5))))`

Reply:
Indeed. An easier way to write these tests that we use in the kernel is to write something like:

    #define GCC_VERSION (__GNUC__ * 10000		\
                         + __GNUC_MINOR__ * 100	\
                         + __GNUC_PATCHLEVEL__)

which can then be compared as an integer.

Reply:
Fixed in https://github.com/lz4/lz4/commit/28e5a4e0542820e81f8372c1a405310e9757cd7e